sql/pkg/repository/memory/memory.go

```diff
@@ -0,0 +1,162 @@
+package memory
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/Mishamba/EPAM_GOLANG_COURSE/sql/pkg/model"
+)
+
+type ContactsRepositoryInMemory struct {
+	storage *sql.DB
+	lastID  uint
+}
+
+func NewContactsRepositoryInMemory() *ContactsRepositoryInMemory {
+	connStr := "user=mishamba password=123 dbname=some-postgres sslmode=disable" //TODO
+	db, _ := sql.Open("postgres", connStr)
+	return &ContactsRepositoryInMemory{
+		storage: db,
+	}
+}
+
+func (r *ContactsRepositoryInMemory) Save(contact model.Contact) (model.Contact, error) {
+	row, err := r.storage.Query("select (Phone, Email) from Contacts")
+	if err != nil {
+		return contact, err
+	}
+
+	defer row.Close()
+	var slice []model.Contact
+	for row.Next() {
+		var tmp model.Contact
+		err := row.Scan(&tmp.ID, &tmp.FirstName, &tmp.LastName, &tmp.Phone, &tmp.Email)
+		if err != nil {
+			return contact, err
+		}
+
+		slice = append(slice, tmp)
+	}
+
+	for _, c := range slice {
+		if c.Email == contact.Email {
+
+			return model.Contact{}, fmt.Errorf("contact with email %q already exists", c.Email)
+		}
+
+		if c.Phone == contact.Phone {
+			return model.Contact{}, fmt.Errorf("contact with phone %q already exists", c.Phone)
+		}
+	}
+
+	r.lastID++
```

Comment:
In memory realisation was concurrency save, there is a race condition in case of concurrent calls to Repository

Reply:
so if there will be no  `lastID`  so we wount have any concurrency. is it correct?